great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -722,7 +741,7 @@ def variables(self, value: Optional[ParameterContainer]):
 
     @property
     def rules(self) -> List[Rule]:
-        return list(self._get_rules_as_dict().values())
+        return list(self.get_rules_as_dict().values())
```

Comment:
Couple of questions:

1. (most important) Why are setting and getting rules using different backing stores?
2. (more philosophical) How/who consumes the rules? Is there an opportunity to encapsulate that work within the rule-based profiler?

Reply:
@donaldheppner 1) Store is not present here as far as I can tell (or did you mean something else by "store"?). 2) The rules are consumed by profiler and also in the case of the self-initializing expectation, exactly one (or zero) rule will be needed (hence the public method).

Reply:
by store I meant different attributes; the "get" property reads from a dictionary and the "setter" initializes `_rules`

Reply:
@donaldheppner Yes, `Rules` are initialized (from config), but they never set in the standard "obj.prop = val" sense (we have not encountered the need for defining a setter, because no external object has required it so far).  There is a `@property` getter, because the rules do need to be obtained by other objects (and tule `RuleBasedProfiler` internal usage, too), and finally, there is a convenience method that returns rules (which are formally a list) as dictionary with the rule name as the key (again, this is a "helper" as a convenience for the needs of other methods).  Thanks.

Reply:
@cdkini -- @donaldheppner and I discussed the need for reconciliation. Also, I was able to make the rules setter "standard" -- please re-review. Thanks!